apps/dashboard/app/(app)/logs/components/filters/components/custom-date-filter.tsx

```diff
@@ -0,0 +1,162 @@
+"use client";
+
+import { addHours, format, setHours, setMinutes, setSeconds } from "date-fns";
+import type { DateRange } from "react-day-picker";
+
+import { Button } from "@/components/ui/button";
+import { Calendar } from "@/components/ui/calendar";
+import { Popover, PopoverContent, PopoverTrigger } from "@/components/ui/popover";
+import { cn } from "@/lib/utils";
+import { ArrowRight, Calendar as CalendarIcon } from "lucide-react";
+import { useEffect, useState } from "react";
+import { useLogSearchParams } from "../../../query-state";
+import TimeSplitInput from "./time-split";
+
+export function DatePickerWithRange({ className }: React.HTMLAttributes<HTMLDivElement>) {
+  const [interimDate, setInterimDate] = useState<DateRange>({
+    from: new Date(),
+    to: new Date(),
+  });
+  const [finalDate, setFinalDate] = useState<DateRange>();
+  const [startTime, setStartTime] = useState({ HH: "09", mm: "00", ss: "00" });
```

Comment:
where do these defaults come from?

Reply:
It's just a random starting hour. Do you have something else on your mind?